quiz.js

```diff
@@ -1,168 +1,137 @@
-            answerZoneEl.append(tempBtn);
+        scoreEl         .text(`Score: ${score}`);
+        questionEl      .text(`Question: ${fix(currentQuestion.question)}`);
+        categoryEl      .text(`Category: ${currentQuestion.category}`);
+
+        let allBtns = createAnswerButtons(currentQuestion)
+            .sort(() => Math.random() -.5);
+            
+        allBtns.forEach(function(btn) {
+            answerZoneEl.append(btn);
+        })
+    }
+
+    function createAnswerButtons(question) {
+        let buttons = []
+        question.incorrect_answers.forEach(function (value) {
+            let tempBtn = $('<button>')
+                .text(fix(value))
+                .val(fix(value))
+                .addClass('hollow button answer')
+                .click(() => validateAnswer(false));
+                buttons.push(tempBtn);
         })
 
+        let correctBtn = $('<button>')
+            .text(question.correct_answer)
+            .val(question.correct_answer)
+            .addClass('hollow button answer')
+            .click(() => validateAnswer(true));
+        buttons.push(correctBtn);
+        return buttons;
     }
-    //this fnction validates the users answer
-    function validateAnswer() {
-        //set userChoice to the value of the btn the user clicked
-        const userChoice = $(this).val();
+
+    function correctAnswerClicked() {
+        score++;
+        rightWrongEl.text(`RIGHT`);
+        if (cheer.pause) {
+            cheer.currentTime = 0;
+            cheer.play();
+        }
+    }
+
+    function inCorrectAnswerClicked() {
+        rightWrongEl.text(`WRONG: Answer is ${correctAnswer}`);
+        if (jeer.pause) {
+            jeer.currentTime = 0;
+            jeer.play();
+        }
+    } 
+
+    //this function validates the users answer
+    function validateAnswer(isCorrect) {
         $(".answer").attr("disabled", true)
-        //if user chooses the correct answer the score goes up 1 and "RIGHT" appears in the right/wrong div
-        if (userChoice === correctAnswer) {
-            score++;
-            rightWrongEl.text(`RIGHT`);
-            if (cheer.pause) {
-                cheer.currentTime = 0;
-                cheer.play();
-            }
-            //if user is wrong "WRONG" apers in the right/wrong div along with the correct answer
+
+        if (isCorrect) { 
+            correctAnswerClicked(); 
         } else {
-            rightWrongEl.text(`WRONG: Answer is ${correctAnswer}`);
-            if (jeer.pause) {
-                jeer.currentTime = 0;
-                jeer.play();
-            }
+            inCorrectAnswerClicked();
         }
-        //question index increases by 1
+
         questionIndex++;
-        //setTimeout function to allow 1 second before generating a new question
-        setTimeout(function () {
-            //calls sendApiRequest function
-            generateQuestion();
+        setTimeout(generateQuestion, 3000)
```

Comment:
since `generateQuestion()` is a function that takes no arguments, we can just pass it in directly here without wrapping it with the function declaration.
```
setTimeout(function() {
    generateQuestion();
}, 3000);

//becomes

setTimeout(generateQuestion, 3000);
```